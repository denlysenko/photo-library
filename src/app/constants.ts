export const FAVORITES_KEY = 'favorites_photos';

export const menuItems = [
  {
    link: '/',
    label: 'Photos',
  },
  {
    link: '/favorites',
    label: 'Favorites',
  },
];
