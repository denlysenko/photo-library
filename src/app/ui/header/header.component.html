<nav class="navbar">
  <a *ngFor="let item of menuItems"
     [routerLink]="[item.link]"
     class="btn btn-basic"
     routerLinkActive="active">{{ item.label }}</a>
</nav>