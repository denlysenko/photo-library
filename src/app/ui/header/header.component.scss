.navbar {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 60px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #1976d2;
  color: white;
  font-weight: 600;
}

.btn-basic {
  width: 150px;
  margin: 8px 16px;

  &.active {
    background-color: rgba(255, 255, 255, 0.15);
    color: white;
    transform: translateY(-3px);
    box-shadow: 0 4px 17px rgba(0, 0, 0, 0.35);
  }
}
