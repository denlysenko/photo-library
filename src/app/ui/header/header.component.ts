import { Component, ChangeDetectionStrategy, Input } from '@angular/core';

import { MenuItem } from '../../interfaces';

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class HeaderComponent {
  @Input() menuItems: MenuItem[];
}
