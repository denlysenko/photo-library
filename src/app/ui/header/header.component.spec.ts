import { CommonModule } from '@angular/common';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { menuItems } from '../../constants';
import { HeaderComponent } from './header.component';

describe('HeaderComponent', () => {
  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [CommonModule, RouterTestingModule],
      declarations: [HeaderComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderComponent);
    component = fixture.componentInstance;
    component.menuItems = menuItems;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture).toMatchSnapshot();
  });
});
