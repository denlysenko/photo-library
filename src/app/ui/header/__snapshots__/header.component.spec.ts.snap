// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should create 1`] = `
<app-header
  menuItems={[Function Array]}
>
  <nav
    class="navbar"
  >
    
    <a
      class="btn btn-basic"
      href="/"
      routerlinkactive="active"
    >
      Photos
    </a>
    <a
      class="btn btn-basic"
      href="/favorites"
      routerlinkactive="active"
    >
      Favorites
    </a>
  </nav>
</app-header>
`;
